### PR TITLE
Add support for `advanced_datapath_observability_config` to `google_container_cluster`

### PR DESCRIPTION
**Pull request by epk**
Adds support for https://cloud.google.com/kubernetes-engine/docs/how-to/configure-dpv2-observability#configure-gke-dataplane-v2-metrics

---

If this PR is for Terraform, I acknowledge that I have:

- [ ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [ ] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
container: added support for `advanced_datapath_observability_config` to `google_container_cluster`
```

### CODE REVIEW

File: mmv1/third_party/terraform/services/container/resource_container_cluster.go.erb [hunk 1]
```diff
@@ -1159,6 +1159,22 @@ func ResourceContainerCluster() *schema.Resource {
 								},
 							},
 						},
+						"advanced_datapath": {
```
Comment (rileykarson) [inline mmv1/third_party/terraform/services/container/resource_container_cluster.go.erb:1162]: Let's mirror the API name exactly, i.e. `advanced_datapath_observability_config`

File: mmv1/third_party/terraform/services/container/resource_container_cluster.go.erb [hunk 2]
```diff
@@ -1159,6 +1159,22 @@ func ResourceContainerCluster() *schema.Resource {
...
+							Type:        schema.TypeList,
+							Optional:    true,
+							Computed:    true,
```
Comment (rileykarson) [inline mmv1/third_party/terraform/services/container/resource_container_cluster.go.erb:1165]: Confirming `Computed` is required, this defaults to:

```
  "advancedDatapathObservabilityConfig": {
   "relayMode": "DISABLED"
  }
```

File: mmv1/third_party/terraform/services/container/resource_container_cluster.go.erb [hunk 3]
```diff
@@ -1159,6 +1159,22 @@ func ResourceContainerCluster() *schema.Resource {
...
+							MaxItems:    1,
+							Description: `Configuration of Advanced Datapath Observability features.`,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
```
Comment (rileykarson) [inline mmv1/third_party/terraform/services/container/resource_container_cluster.go.erb:1169]: Should we add `relay_mode` at the same time? https://cloud.google.com/kubernetes-engine/docs/reference/rest/v1beta1/projects.locations.clusters#Cluster.AdvancedDatapathObservabilityConfig

File: mmv1/third_party/terraform/tests/resource_container_cluster_test.go.erb [hunk 4]
```diff
@@ -7475,6 +7493,36 @@ resource "google_container_cluster" "primary" {
...
 }
 
+func testAccContainerCluster_withMonitoringConfigAdvancedDatapathMetricsEnabled(name string) string {
+       return fmt.Sprintf(`
+resource "google_container_cluster" "primary" {
+  name               = "%s"
+  location           = "us-central1-a"
+  initial_node_count = 1
+  monitoring_config {
+		advanced_datapath {
```
Comment (rileykarson) [inline mmv1/third_party/terraform/tests/resource_container_cluster_test.go.erb:7503]: nit: these multiline strings are generally two-spaced, while the Go portions of the file are tab-spaced. Mind cleaning up the spacing here & below? You can see this snippet mixes them.

**Comment by rileykarson** [comment]
From the failing build:

```
==> Checking that code complies with gofmt requirements...
go vet
go: cloud.google.com/go/compute@v1.1[9](https://github.com/GoogleCloudPlatform/magic-modules/actions/runs/5591925258/jobs/10223680551#step:9:10).3: missing go.sum entry for go.mod file; to add it:
	go mod download cloud.google.com/go/compute
make: *** [GNUmakefile:26: vet] Error 1
Error: Process completed with exit code 2.
```

**Comment by epk** [comment]
@rileykarson should be good to test/review. Thanks!

**Comment by epk** [comment]
Rebased on main

**Pull request closed without merging**